server/controllers/changeNonResidentialType/details.ts

```diff
@@ -44,6 +45,29 @@ export default class ChangeNonResidentialTypeDetails extends FormInitialStep {
     }
   }
 
+  async validateFields(req: FormWizard.Request, res: Response, callback: (errors: any) => void) {
+    super.validateFields(req, res, async errors => {
+      const { location } = res.locals
+      const { prisonId, id: locationId } = location
+
+      const { convertedCellType } = req.form.values
+      const { otherConvertedCellType } = req.form.values
+      const currentConvertedCellType = res.locals.location.raw.convertedCellType
+      const currentOtherConvertedCellType = res.locals.location.raw.otherConvertedCellType
+
+      const convertedCellTypeUnchanged = convertedCellType === currentConvertedCellType
+      const otherConvertedCellTypeUnchanged = otherConvertedCellType === currentOtherConvertedCellType
+
+      if (
+        (convertedCellTypeUnchanged && convertedCellType !== 'OTHER') ||
+        (convertedCellType === 'OTHER' && convertedCellTypeUnchanged && otherConvertedCellTypeUnchanged)
+      ) {
+        return res.redirect(`/view-and-update-locations/${prisonId}/${locationId}`)
+      }
+      return callback({ ...errors })
+    })
+  }
+
```

Comment:
Alongside returning errors, validation redirects if no changes have been made. If changes have, we show success banner.